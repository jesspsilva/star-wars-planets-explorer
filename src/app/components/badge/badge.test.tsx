import { render, screen } from "@testing-library/react";

import Badge from "./badge";

describe("Badge component", () => {
  it("matches snapshot", () => {
    const { container } = render(<Badge label="Badge" />);
    expect(container).toMatchSnapshot();
  });

  it("should render with default colors", () => {
    render(<Badge label="Badge" />);

    const badge = screen.getByTestId("badge");
    expect(badge).toHaveStyle({
      backgroundColor: "#b5b7c0",
      border: "1px solid #b5b7c0",
      color: "#b5b7c0",
    });
  });

  it("should render with custom colors when provided", () => {
    render(
      <Badge
        label="Badge"
        colors={{
          background: "BG_COLOR",
          border: "BORDER_COLOR",
        }}
      />
    );

    const badge = screen.getByTestId("badge");
    expect(badge).toHaveStyle({
      backgroundColor: "BG_COLOR",
      border: "1px solid BORDER_COLOR",
      color: "BORDER_COLOR",
    });
  });

  it("should display custom label", () => {
    render(<Badge label="Badge Label" />);

    const badge = screen.getByTestId("badge");
    expect(badge).toHaveTextContent("Badge Label");
  });
});
