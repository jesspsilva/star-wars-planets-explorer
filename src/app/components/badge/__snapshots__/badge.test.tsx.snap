// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge component matches snapshot 1`] = `
<div>
  <span
    class="sc-blHHSb kYIjTj"
    colors="[object Object]"
    data-testid="badge"
    label="Badge"
  >
    Badge
  </span>
</div>
`;
